pkg/controllers/user/logging/clusterlogging.go

```diff
@@ -113,3 +127,43 @@ func (c *ClusterLoggingSyncer) createOrUpdateClusterConfigMap() error {
 	}
 	return utils.UpdateConfigMap(loggingconfig.ClusterConfigPath, loggingconfig.ClusterLoggingName, "cluster", c.configmaps)
 }
+
+func updateEmbeddedEndpoint(podLister v1.PodLister, serviceLister v1.ServiceLister, clusterLoggings v3.ClusterLoggingInterface) {
+	for {
+		select {
+		case <-timeout:
+			logrus.Warn("update embedded endpoint timeout")
+			return
+		case <-tick:
+			cls, err := clusterLoggings.List(metav1.ListOptions{})
+			if err != nil {
+				logrus.Debugf("get cluterlogging failed, %v", err)
+			}
+			if len(cls.Items) != 0 {
+				cl := cls.Items[0]
+				if cl.Spec.EmbeddedConfig != nil && (cl.Spec.EmbeddedConfig.ElasticsearchEndpoint == "" || cl.Spec.EmbeddedConfig.KibanaEndpoint == "") {
```

Comment:
IIUC, The endpoints are updated only when any of the endpoint fields is empty. Is it possible that the endpoint is changed?(like nodeIP that we got previously becomes invalid).